### PR TITLE
Random

### PR DESCRIPTION
**Pull request by IuriyLeb**

### CODE REVIEW

File: Randomizer.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
+import random
+
+
+def randomly(list_of_students):
+    student_dict_reverted = {}
+    for elem in list_of_students:
+        student_dict_reverted[elem] = 0
```
Comment (AleksandrSl) [inline Randomizer.py]: Why this is called `reverted`?

File: Randomizer.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import random
+
+
+def randomly(list_of_students):
+    student_dict_reverted = {}
+    for elem in list_of_students:
+        student_dict_reverted[elem] = 0
+    print('Do you want to make your choice?', '\n', 'Press y to do it, n to exit')
+    while True:
+        choice2 = input()
```
Comment (AleksandrSl) [inline Randomizer.py]: Why `choice2`, not `choice1` or `choice9000`

File: Randomizer.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+    while True:
+        choice2 = input()
+        if choice2 == 'n':
+            break
+        elif choice2 == 'y':
+            somestud = random.choice(list_of_students)
+            if student_dict_reverted[somestud] > 0:
+                somestud = random.choice(list_of_students)
+                student_dict_reverted[somestud] = 0
+                print(somestud)
```
Comment (AleksandrSl) [inline Randomizer.py]: Repeat of prints and `student_dict_reverted[somestud] += 1` can be avoided by moving them outside condition statement

File: Randomizer.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import random
+
+
+def randomly(list_of_students):
```
Comment (AleksandrSl) [inline Randomizer.py]: Function names should be verb, not adverb

File: Randomizer.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+import random
+
+
+def randomly(list_of_students):
+    student_dict_reverted = {}
+    for elem in list_of_students:
+        student_dict_reverted[elem] = 0
+    print('Do you want to make your choice?', '\n', 'Press y to do it, n to exit')
```
Comment (AleksandrSl) [inline Randomizer.py]: Maybe ask whether we want to peek one more student at the end of while loop. It's a strange situation when we run our script and then entered `n`.

**Review by AleksandrSl — CHANGES_REQUESTED** [review]

File: Randomizer.py (hunk 1 above)
Comment (IuriyLeb) [reply Randomizer.py]: Remained from the old code, already fixed.

File: Randomizer.py (hunk 2 above)
Comment (IuriyLeb) [reply Randomizer.py]: Remained from the old code, already fixed[2]

File: Randomizer.py (hunk 3 above)
Comment (IuriyLeb) [reply Randomizer.py]: Did not notice this.
Thanks, corrected.

File: Randomizer.py (hunk 4 above)
Comment (IuriyLeb) [reply Randomizer.py]: Done:)

File: Randomizer.py (hunk 5 above)
Comment (IuriyLeb) [reply Randomizer.py]: Students should be selected constantly, I see no reason to make a confirmation after each choice.

Comment (AleksandrSl) [reply Randomizer.py]: However I have to confirm that I want to make a new choice 
```
python3 test.py 
Do you want to make your choice? 
 Press y to do it, n to exit
y
D
y
B
y
B
y
B
```

Comment (AleksandrSl) [reply Randomizer.py]: I suggest to consider something like this:
```
D
Proceed? (y/n) y
B
Proceed? (y/n)  y
B
Proceed? (y/n) y
B
```

File: Randomizer.py [hunk 6]
```diff
@@ -1,24 +1,25 @@
 import random
 
 
-def randomly(list_of_students):
-    student_dict_reverted = {}
+def random_student(list_of_students):
```
Comment (AleksandrSl) [inline Randomizer.py]: Better, but still it's not an action)

File: Randomizer.py [hunk 7]
```diff
@@ -1,24 +1,25 @@
...
-    print('Do you want to make your choice?', '\n', 'Press y to do it, n to exit')
+        student_dict[elem] = 0
+    # print('Do you want to make your choice?', '\n', 'Press y to do it, n to exit')
+    somestud = random.choice(list_of_students)
+    print(somestud)
+    student_dict[somestud] += 1
     while True:
-        choice2 = input()
-        if choice2 == 'n':
+        print('Proceed? (y/n)', end=' ')
```
Comment (AleksandrSl) [inline Randomizer.py]: Forgot to mention this before, read documentation about `input()`, it takes some arguments that may alleviate you pain.

File: Randomizer.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import random
+
+
+def random_student(list_of_students):
+    student_dict = {}
+    for elem in list_of_students:
+        student_dict[elem] = 0
+    # print('Do you want to make your choice?', '\n', 'Press y to do it, n to exit')
```
Comment (agbragin) [inline Randomizer.py]: Commented code.

File: Randomizer.py [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+def random_student(list_of_students):
+    student_dict = {}
+    for elem in list_of_students:
+        student_dict[elem] = 0
+    # print('Do you want to make your choice?', '\n', 'Press y to do it, n to exit')
+    somestud = random.choice(list_of_students)
+    print(somestud)
+    student_dict[somestud] += 1
+    while True:
+        print('Proceed? (y/n)', end=' ')
```
Comment (agbragin) [inline Randomizer.py]: We've decided to extract all code for interaction with the user into the common code.

File: Randomizer.py [hunk 10]
```diff
@@ -0,0 +1,25 @@
+import random
+
+
+def random_student(list_of_students):
+    student_dict = {}
+    for elem in list_of_students:
+        student_dict[elem] = 0
+    # print('Do you want to make your choice?', '\n', 'Press y to do it, n to exit')
+    somestud = random.choice(list_of_students)
+    print(somestud)
```
Comment (agbragin) [inline Randomizer.py]: Do we actually need the `print` here?

File: Randomizer.py [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    student_dict[somestud] += 1
+    while True:
+        print('Proceed? (y/n)', end=' ')
+        choice = input()
+        if choice == 'n':
+            break
+        elif choice == 'y':
+            somestud = random.choice(list_of_students)
+            if student_dict[somestud] > 0:
```
Comment (agbragin) [inline Randomizer.py]: Could be simplified to `if student_dict[somestud]`.

File: Randomizer.py [hunk 12]
```diff
@@ -0,0 +1,25 @@
+import random
+
+
+def random_student(list_of_students):
```
Comment (agbragin) [inline Randomizer.py]: The doc comment here could help to get the idea.

**Review by agbragin — CHANGES_REQUESTED** [review]
Let's move the user interaction code into the common part.

**Pull request closed without merging**